app/components/HighlighterModule/highlighterModule.js

```diff
@@ -0,0 +1,437 @@
+        blue += fraction * Number(rgb[2]);
+        index += 1;
+      }
+      var opacity = 0.5;
+      if (selected) {
+        opacity = 1;
+      }
+      if (list.length == 0) {
+        return 'rgba(255, 255, 255, 0)';
+      }
+      return 'rgba(' + Math.round(red) + ', ' + Math.round(green) + ', ' + Math.round(blue) + ', ' + opacity +')';
+
+    }
+
+    function processHighlights() {
+      var highlights = privateHighlights;
+      var topics = privateTopics.results;
+      var selectedHighlights = privateSelected;
+      var parsedHighlights = [];
+      var final = [];
+      var temp_index = 0;
+      while (temp_index < highlights.length) {
+        var i = highlights[temp_index];
+        var start = {type: 'start', index: i.start, topic: i.topic, source: i, selected: false};
+        var end = {type: 'end', index: i.end, topic: i.topic, source: i, selected: false};
+        parsedHighlights.push(start);
+        parsedHighlights.push(end);
+        temp_index += 1;
+      }
+
+      // (2)
+      parsedHighlights.sort((a,b) => {
+        return a.index - b.index;
+      });
+
+      var activeSources = [];
+      var topicNum = topics.length;
+      var activeTopics = [];
+      var topic_list = [];
+      for (i=0; i<topicNum; i++) {
+        activeTopics.push(false);
+        topic_list.push(topics[i].id)
+      }
+      var activeSelect = false;
+      var start = 0;
+      var end = 0;
+      temp_index = 0;
+
+      // (3)
+      while (temp_index < parsedHighlights.length) {
+        var i = parsedHighlights[temp_index];
+        var processed = {start: null, end: null, topics: [], source: activeSources.slice(0), selected: false};
+        processed.start = start;
+        processed.end = i.index;
+
+        // (4)
+        if (selectedHighlights.length) {
+          var select_index = 0;
+          while (select_index < selectedHighlights.length) {
+
+            var selected_high = selectedHighlights[select_index]
+            if ((selected_high[0] == processed.start) && (selected_high[1] == processed.end)) {
+              processed.selected = true;
+              break;
+            } else if ((selected_high[0] < processed.start) && (processed.start < selected_high[1])) {
+              processed.selected = true;
+              break;
+            } else if ((selected_high[0] < processed.end) && (processed.end < selected_high[1])) {
+              processed.selected = true;
+              break;
+            }
+            select_index += 1;
+          }
+        }
+
+        start = i.index;
+        var list_index = 0;
+        while (list_index < activeTopics.length) {
+          if (activeTopics[list_index]) {
+            processed.topics.push(topic_list[list_index]);
+          }
+          list_index += 1;
+        }
+        final = final.concat(processed);
+
+        // (5) Activate/Deactivate Topics
+        var active_state = i.type === 'start'
+        for(index=0;index<activeTopics.length;index+=1) {
```

Comment:
this module should have no concept of topics. it should only understand the concept of a highlight and a highlight color.